packages/data/src/redux-store/index.js

```diff
@@ -290,18 +291,24 @@ export default function createReduxStore( key, options ) {
 			const subscribe =
 				store &&
 				( ( listener ) => {
-					let lastState = store.__unstableOriginalGetState();
-					return store.subscribe( () => {
-						const state = store.__unstableOriginalGetState();
-						const hasChanged = state !== lastState;
-						lastState = state;
-
-						if ( hasChanged ) {
-							listener();
-						}
-					} );
+					privateListeners.add( listener );
+
+					return () => privateListeners.delete( listener );
 				} );
 
+			let lastState = store.__unstableOriginalGetState();
+			store.subscribe( () => {
+				const state = store.__unstableOriginalGetState();
+				const hasChanged = state !== lastState;
+				lastState = state;
+
+				if ( hasChanged ) {
+					for ( const listener of privateListeners ) {
+						listener();
+					}
+				}
+			} );
```

Comment:
So, we are keeping the per-store subscriptions, and only optimize the `prevState === nextState` check? Doing it just once instead of X times?

Reply:
yeah, it seems like this _could_ be the key piece of this change, since Redux itself is doing basically the same think with a list of listeners.

I don't fully know why this is effective, but it seems to measure that way.

Reply:
In an active editor, there are 1000 listeners, and every keystroke, when typing, generates 1000 calls. It's plausible that even such a small operation, on code that doesn't do anything expensive (getter call + `===` compare), has a visible impact when removed.

By the way, do you have raw timing data, i.e., all the individual measurements instead of only averages, for these 180 rounds you did? It would be interesting to calculate the variances, how the errors are big and how they are distributed.

Reply:
I don't have that information, and while it's possible to gather it, it can be meticulous. The `min` and `max` values give us some first-order approximation of the variance.

At some point I'd like to save all the values by default. Running multiple `--rounds` is much easier.